threads.go

```diff
@@ -205,11 +205,19 @@ func (t *threads) Put(ctx context.Context, rec thread.Record, opts ...tserv.PutO
 func (t *threads) Pull(ctx context.Context, id thread.ID, lid peer.ID, offset cid.Cid, opts ...tserv.PullOption) ([]thread.Record, error) {
 	settings := tserv.PullOptions(opts...)
 
-	if t.PubKey(id, lid) == nil {
+	pk, err := t.PubKey(id, lid)
+	if err != nil {
+		return nil, fmt.Errorf("couldn't fetch public key from book: %v", err)
```

Comment:
After looking at the books, it looks like you are already wrapping these errors. Do you think we need the additional wrap here? It just makes the code less readable IMO.

Reply:
Yes, you're right. I think since I already wrap them in the books, it could be returned _as is_ at this level.

Regarding errors in general, I had pending a bit more thinking about this. Go 1.13 already has a stronger opinion about error wrapping and some new primitives to make life easier (since this string returning stuff leaves no room for the caller to check anything). 

For the moment I'm going to wipe off these `fmt.Errorf`s that are wrapped in books, and then consolidate better what I'm thinking about the error-style thing to chat about it.